DataProcessing/datax-host/src/main/scala/datax/host/BlobBatchingHost.scala

```diff
@@ -7,47 +7,65 @@ package datax.host
 import java.sql.Timestamp
 import java.text.SimpleDateFormat
 import java.time.Instant
-import java.util.concurrent.Executors
+import java.time.temporal.ChronoUnit
 
 import datax.config.UnifiedConfig
 import datax.constants.ProductConstant
+import datax.data.FileInternal
 import datax.fs.HadoopClient
 import datax.input.BatchBlobInputSetting
-import datax.processor.BlobPointerProcessor
+import datax.processor.{BatchBlobProcessor, CommonProcessorFactory}
 import datax.telemetry.AppInsightLogger
-import datax.utility.DataMerger
 import org.apache.log4j.LogManager
 
 import scala.language.postfixOps
 import scala.collection.mutable.{HashSet, ListBuffer}
-import scala.collection.parallel.ExecutionContextTaskSupport
-import scala.concurrent.ExecutionContext
 import scala.concurrent.duration._
 
 object BlobBatchingHost {
-  def getInputBlobPathPrefixes(prefix: String, datetimeFormat: String, startTime: Instant, durationInSeconds: Long, timeIntervalInSeconds: Long):Iterable[(String, Timestamp)] = {
+  val appLog = LogManager.getLogger("runBatchApp")
+
+ def getInputBlobPathPrefixes(path: String, startTime: Instant, processingWindowInSeconds: Long, partitionIncrementDurationInSeconds: Long):Iterable[(String, Timestamp)] = {
     val result = new ListBuffer[(String, Timestamp)]
     val cache = new HashSet[String]
 
+    val datetimeFormat = getDateTimePattern(path)
     var t:Long = 0
-    //val utcZoneId = ZoneId.of("UTC")
-    val dateFormat = new SimpleDateFormat(datetimeFormat)
-    while(t<durationInSeconds){
-      val timestamp = Timestamp.from(startTime.plusSeconds(t))
-      val partitionFolder = dateFormat.format(timestamp)
-      if(!cache.contains(partitionFolder)){
-        val path = prefix+partitionFolder
-        result += Tuple2(path, timestamp)
-        cache += partitionFolder
+
+    if(!datetimeFormat.isEmpty) {
+      val dateFormat = new SimpleDateFormat(datetimeFormat)
+      while (t <= processingWindowInSeconds) {
+        val timestamp = Timestamp.from(startTime.plusSeconds(t))
+        val partitionFolder = dateFormat.format(timestamp)
+        val path2 = path.replace("{" + datetimeFormat + "}", partitionFolder)
+        if (!cache.contains(partitionFolder)) {
+          result += Tuple2(path2, timestamp)
+          cache += partitionFolder
+        }
+        t += partitionIncrementDurationInSeconds
       }
-      t+= timeIntervalInSeconds
+    }
+    else {
+      result += Tuple2(path, Timestamp.from(Instant.now()))
     }
 
     result
   }
 
-  def runBatchApp(inputArguments: Array[String],processorGenerator: UnifiedConfig=>BlobPointerProcessor ) = {
-    val appLog = LogManager.getLogger("runBatchApp")
+  // Get the datetime pattern like {yyyy-MM-dd} in the input path.
+  // For e.g. if the input path is "wasbs://outputs@myaccount.blob.core.windows.net/{yyyy-MM-dd}/flow1", this will return yyyy-MM-dd
+  private def getDateTimePattern(inputPath:String):String ={
+    val regex = """\{([yMdHmsS\-/.]+)\}*""".r
```

Comment:
Can we considering caching this pattern probably as a member of BlobBatchingHost object, or some other common place if this regex could be used for other parts of the code